pkg/completion/completion.go

```diff
@@ -95,6 +95,9 @@ type Completion struct {
 	// ctx is the context of the wait group.
 	ctx context.Context
 
+	// lock is used to check and close the completed channel atomically.
+	lock lock.Mutex
```

Comment:
If this is directly nested in the structure, you can avoid the `foo.lock.Lock()` verbosity and just do `foo.Lock()`, any reason not to do this?

Reply:
I didn't want the mutex to be exported.